library/src/main/java/com/google/android/exoplayer/extractor/ts/MpaReader.java

```diff
@@ -0,0 +1,293 @@
+        // Do nothing.
+    }
+
+    /**
+     * Continues a read from the provided {@code source} into a given {@code target}. It's assumed
+     * that the data should be written into {@code target} starting from an offset of zero.
+     *
+     * @param source The source from which to read.
+     * @param target The target into which data is to be read.
+     * @param targetLength The target length of the read.
+     * @return Whether the target length was reached.
+     */
+    private boolean continueRead(ParsableByteArray source, byte[] target, int targetLength) {
+        int bytesToRead = Math.min(source.bytesLeft(), targetLength - bytesRead);
+        source.readBytes(target, bytesRead, bytesToRead);
+        bytesRead += bytesToRead;
+        return bytesRead == targetLength;
+    }
+
+    /**
+     * Locates the next sync word, advancing the position to the byte that immediately follows it.
+     * If a sync word was not located, the position is advanced to the limit.
+     *
+     * @param pesBuffer The buffer whose position should be advanced.
+     * @return True if a sync word position was found. False otherwise.
+     */
+    private boolean skipToNextSync(ParsableByteArray pesBuffer) {
+        byte[] mpaData = pesBuffer.data;
+        int startOffset = pesBuffer.getPosition();
+        int endOffset = pesBuffer.limit();
+        for (int i = startOffset; i < endOffset - 1; i++) {
+            int syncBits = ((mpaData[i] & 0xFF) << 8 ) | (mpaData[i + 1] & 0xFF);
+            if ((syncBits & 0xFFF0) == 0xFFF0) {
+                hasCrc = (mpaData[i + 1] & 0x1) == 0;
+                pesBuffer.setPosition(i);
+                return true;
+            }
+        }
+        pesBuffer.setPosition(endOffset);
+        return false;
+    }
+
+    /**
+     * Calculates MPEG Audio frame size
+     *
+     * @param layer The MPEG layer
+     * @param LSF Low Sample rate Format (MPEG 2)
+     * @param bitrate The bitrate in bits per second
+     * @param samplesPerSec The sampling rate in hertz
+     * @param -paddingSize
+     * @return Frame size in bytes
+     */
+    private static int CalcMpaFrameSize (int layer, int LSF, int bitrate, int samplesPerSec, int paddingSize) {
+        return (int)(Math.floor(MPA_COEFFICIENTS[LSF][layer] * bitrate / samplesPerSec) + paddingSize) * MPA_SLOT_SIZE[layer];
+    }
+
+    /**
+     * Parses the sample header.
+     */
+    private void parseHeader() {
+        int headerLength = hasCrc ? HEADER_SIZE + CRC_SIZE : HEADER_SIZE;
+
+        if (!hasOutputFormat) {
+            mpaScratch.setPosition(0);
+            mpaScratch.skipBits(12);
+            int isLSF = (!mpaScratch.readBit()) ? 1 : 0;
+            int layer = mpaScratch.readBits(2) ^ 3;
+            mpaScratch.skipBits(1);
+            int audioObjectType = 32 + layer;
+            int bitRate = MPA_BITRATES[isLSF][layer][mpaScratch.readBits(4)];
+            int sampleRate = MPA_SAMPLING_RATES[3 - isLSF][mpaScratch.readBits(2)];
```

Comment:
because the audio source is from FME live stream, i can only provide fragment of a frame. 

ff e3 20 70 b8 0b e8 8b 67 1f ce 20 00 01 20 02 5c ad 80 00 00 e9 ff d1 69 f4 6d 7a 96 ee d6 ed 3c 6f ff ff c0 00 00 ad 34 49 35 55 5b 6a 5c db 96 c6 c0 07 2d 40 6f e8 c1 c7 a8 71 58 cf 31 86

it seem that the header is correct! 

http://www.mp3-tech.org/programmer/frame_header.html